delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/OrganizationsRoutes.scala

```diff
@@ -49,9 +56,30 @@ final class OrganizationsRoutes(identities: Identities, organizations: Organizat
       OrganizationSearchParams(deprecated, rev, createdBy, updatedBy)
     }
 
+  private def authorizeForOrgUUID(uuid: UUID, permission: Permission)(implicit
+      caller: Caller
+  ): Directive1[OrganizationResource] =
+    onSuccess(organizations.fetch(uuid).runToFuture).flatMap {
+      case Some(org) => authorizeFor(AclAddress.Organization(org.value.label), permission).tmap(_ => org)
+      case None      =>
```

Comment:
at the same time it's quite confusing if you get 403, when something is not found. It would leak whether an organization with a given UUID exists, but I'm not sure how big of a problem that is.

Reply:
it would be the same behaviour as fetching something by label...  authorization happens before checking for existence.

Reply:
That makes sense.